src/listener/handlers/document-update-handler.ts

```diff
@@ -152,32 +150,35 @@ class DocumentUpdateHandler implements IHandler {
     checkEligibleForFileValidation(contentValidationReport: ValidationReport, documentUrl: string, documentType: string): Promise<ValidationReport> {
         if (documentType.toUpperCase() === 'FILE') {
             // refresh doc before checking cloudUrl
-            return this.ingestClient.retrieveMetadataDocument(documentUrl).then((doc: any) => {
-                if (doc['cloudUrl']) {
-                    // if cloud url present, proceeds with file validation
-                    if (contentValidationReport.validationState.toUpperCase() == "VALID") {
-                        return this.attemptFileValidation(contentValidationReport, doc, documentType)
+            return this.ingestClient.retrieveMetadataDocument(documentUrl)
+                .then((doc: any) => {
+                    if (doc['cloudUrl']) {
+                        // if cloud url present, proceeds with file validation
+                        if (contentValidationReport.validationState.toUpperCase() == "VALID") {
+                            return this.attemptFileValidation(contentValidationReport, doc)
+                        }
+                    } else {
+                        return this.validateFileFormat(contentValidationReport, doc).then(
+                            contentValidationReport => {
+                                // otherwise set validation state to INVALID, return error report with NoCloudUrl
+                                const msg = "File cloudUrl property not set.";
+                                const err = new ErrorReport(ErrorType.FileNotUploaded, msg);
+                                err.userFriendlyMessage = "File not uploaded.";
+
+                                contentValidationReport.validationState = "INVALID";
+
+                                if (!contentValidationReport.validationErrors || contentValidationReport.validationErrors.length == 0) {
```

Comment:
small thing but you can just do `!contentValidationReport.validationErrors.length` since `0` evaluates to falsey